### PR TITLE
Count occurences in the rejection reasons

### PR DESCRIPTION
**Pull request by kamsi**
https://unctad.atlassian.net/browse/EV2CORE-2

We will need to prepare a script which calculates it and saves it. The script should operate only in scope of mongo, can be launched at start of the server.

We want to group by rejection reasons (data from first row) and by date.

### CODE REVIEW

File: server/services/recalculate-rejection-reasons-occurrences.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+					}
+				}
+			]).toArray();
+		}).then(function (result) {
+			return deferred.map(result, function (item) {
+				if (item.count <= 1) return;
+				return collection.update({ 'date.date': item._id.date,
+					rejectionReasonsConcat: item._id.rejectionReasonsConcat
+					},
+					{ $set: { occurrencesCount: item.count } });
```
Comment (roxanadina) [inline server/services/recalculate-rejection-reasons-occurrences.js]: and @kamsi third param: {multi:true}
https://docs.mongodb.com/manual/reference/method/db.collection.update/#multi-parameter

File: server/services/recalculate-rejection-reasons-occurrences.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
+'use strict';
+
+var mongo    = require('../mongo-db')
+  , Set      = require('es6-set')
+  , deferred = require('deferred');
+
+module.exports = function () {
```
Comment (mtuchowski) [inline server/services/recalculate-rejection-reasons-occurrences.js]: If I understand correctly this is supposed to be a one time migration right?

Comment (kamsi) [inline server/services/recalculate-rejection-reasons-occurrences.js]: @mtuchowski no it's not. It is supposed to update `rejectionReasonsConcat` on every server start in order to facilitate grouping query search.

Check out the adjusts...

**Comment by mtuchowski** [comment]
OK, waiting for adjusts to be approved.

**Pull request merged**